src/v/utils/delta_for.h

```diff
@@ -94,37 +160,36 @@ static constexpr uint32_t FOR_buffer_depth = 16;
  * The compressed data is stored internally using an iobuf. This iobuf
  * can be copied and pushed to the decoder to decompress the values.
  */
-template<class TVal>
+template<class TVal, class DeltaStep = details::delta_xor>
```

Comment:
Would be great if the preceding comment were modified with an explanation of what this `DeltaStep` class is supposed to do (right now it just states that the values are encoded using xor encoding).